module/documents/actor/actor.mjs

```diff
@@ -1526,6 +1546,24 @@ export default class Actor5e extends Actor {
 
   /* -------------------------------------------- */
 
+  /** @inheritdoc */
+  async rollInitiative(options={}) {
+    const combat = await super.rollInitiative(options);
+    const combatant = this.isToken ? combat.getCombatantByToken(this.token.id) : combat.getCombatantByActor(this.id);
```

Comment:
The `Actor#rollInitiative` method may create initiative rolls for multiple combatants, not necessarily only one.